ush/ioda/bufr2ioda/run_bufr2ioda.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+import argparse
+import os
+from gen_bufr2ioda_json import gen_bufr_json
+from wxflow import Logger, Executable, cast_as_dtype
+
+# Initialize root logger
+logger = Logger('run_bufr2ioda.py', level='INFO', colored_log=True)
+
+
+def bufr2ioda(CDATE, RUN, DMPDIR, config_template_dir, COM_OBS):
+    logger.info(f"Process {CDATE} {RUN} from {DMPDIR} to {COM_OBS} using {config_template_dir}")
+
+    # Derived parameters
+    PDY = CDATE[:8]
+    cyc = CDATE[8:10]
+
+    # Load configuration
+    config = {
+        'RUN': RUN,
+        'PDY': PDY,
+        'cyc': cyc,
+        'DMPDIR': DMPDIR,
+        'COM_OBS': COM_OBS
+    }
+    config['current_cycle'] = cast_as_dtype(CDATE)
+
+    # Get gdasapp root directory
+    DIR_ROOT = os.path.realpath(os.path.join(os.path.dirname(os.path.realpath(__file__)), "../../.."))
+    USH_IODA = os.path.join(DIR_ROOT, "ush", "ioda", "bufr2ioda")
+
+    # Create output directory if it doesn't exist
+    os.makedirs(COM_OBS, exist_ok=True)
+
+    # Specify observation types to be processed by a script
+    BUFR_py = ["satwind_amv_goes"]
+
+    for obtype in BUFR_py:
+        logger.info(f"Convert {obtype}...")
+        json_output_file = os.path.join(COM_OBS, f"{obtype}_{PDY}{cyc}.json")
+        template = config_template_dir + '/bufr2ioda_' + obtype + '.json'
+        gen_bufr_json(config, template, json_output_file)
+
+        # Use the converter script for the ob type
+        bufr2iodapy = USH_IODA + '/bufr2ioda_' + obtype + ".py"
+        cmd = Executable(bufr2iodapy)
+        cmd.add_default_arg('-c')
+        cmd.add_default_arg(json_output_file)
+        logger.info(f"Executing {cmd}")
+        cmd()
```

Comment:
If this is a python script, then why do we need to step out of the interpreter into a subprocess?

Reply:
Agreed.  

In principal we should not & do not want to step out of the python script via a subprocess to invoke another python script.  However, at present each observation type has specific scripting to generate an ioda format dump file the source bufr dump file.   A python [factory](https://github.com/NOAA-EMC/GDASApp/pull/601#discussion_r1316262132) may be the best way to tackle this.  This noted, a python converter may not be the long term solution.  We may opt to create ioda format dumps via c++ code.  This c++ code might eventually move into JEDI as a bufr to ioda interface. 

The _python scripts_ approach introduced by this PR along with g-w PR [#1826](https://github.com/NOAA-EMC/global-workflow/pull/1826) are needed for JEDI ATM prototyping.   We want to move away from the current approach of (a) run `gsi.x` in observer mode to create netcdf diagnostic files followed by (b) convert `gsi.x` netcdf diagnostic files into ioda format observation files.

Reply:
You may want to consider looking into the `importlib` module.  It provides the same functionality as `import` does at the top of the script, but is designed for this particular type of situation you are in where you do not want to import everything at the top.

Reply:
Thank you @aerorahul for this suggestion.  Something to consider in future revisions.